simapp/v2/simdv2/cmd/testnet.go

```diff
@@ -402,6 +403,11 @@ func initGenFiles[T transaction.Tx](
 	}
 	appGenState[banktypes.ModuleName] = clientCtx.Codec.MustMarshalJSON(&bankGenState)
 
+	var bankV2GenState bankv2types.GenesisState
+	clientCtx.Codec.MustUnmarshalJSON(appGenState[bankv2types.ModuleName], &bankV2GenState)
```

Comment:
Why we need this step as you are assigning `bankV2GenState` a new value in next step?

Reply:
yes should remove it

Reply:
I like the fallback tbh fpr mpw, maybe we should check `if len(bankV2GenState.Balances) > 0` and if not, we use bank v1 genesis?